main.tf

```
@@ -52,7 +52,7 @@ locals {
   redis_multishard = "${local.elasticache_name == "redis" && var.redis_multi_shard ? true : false}"
 
   # Default for number_of_nodes is 1, but at least 2 is required for redis non-multi-shard
-  redis_node_count = "${var.number_of_nodes < 2 ? 2 : var.number_of_nodes}"
+  redis_node_count = "${var.number_of_nodes < 2 && var.failover_enabled ? 2 : var.number_of_nodes}"
```

@mlattanzi In line with my comments in `#faws-support-engineering`, I think it'd be better to force an error here instead of silently switching to `2` b/c principle of least surprise. Lacking a clean way to display specific error messages, using `-1` for the node count should cause the whole thing to blow up and call attention to the invalid configuration at least.
```suggestion
  redis_node_count = "${var.number_of_nodes < 2 && var.failover_enabled ? -1 : var.number_of_nodes}"
```